cristin-import/src/main/java/no/unit/nva/cristin/mapper/CristinMapper.java

```diff
@@ -102,16 +105,35 @@ private Organization extractOrganization() {
         return new Organization.Builder().withId(customerId.getUri()).build();
     }
 
-    private Instant extractEntryCreationDate() {
+    private Instant extractDate() {
         return Optional.ofNullable(cristinObject.getEntryCreationDate())
-                   .map(ld -> ld.atStartOfDay().toInstant(zoneOffset()))
+                   .map(this::localDateToInstant)
+                   .orElseGet(() -> extractPublishedDateOrConvertYearToInstant(cristinObject));
+    }
+
+    private Instant extractPublishedDateOrConvertYearToInstant(CristinObject cristinObject) {
+        return Optional.ofNullable(cristinObject.getEntryPublishedDate())
+                   .map(this::localDateToInstant)
+                   .orElseGet(() -> convertPublishedYearInstant(cristinObject));
+    }
+
+    private Instant convertPublishedYearInstant(CristinObject cristinObject) {
+
+        return Optional.ofNullable(cristinObject.getPublicationYear())
+                   .map(year -> LocalDate.of(year, Month.JANUARY, FIRST_DAY_OF_MONTH)
+                                    .atStartOfDay()
+                                    .toInstant(zoneOffset()))
```

Comment:
Kanskje lage metode toInstant eller lignende?

                   .map(this::toInstant)
                   .orElse(null);
    }

    private Instant toInstant(Integer year) {
        return LocalDate.of(year, Month.JANUARY, FIRST_DAY_OF_MONTH)
                   .atStartOfDay()
                   .toInstant(zoneOffset());
    }

Reply:
Fikset